syntax/vim/syntax/assembly.vim

```diff
@@ -0,0 +1,79 @@
+" Vim syntax file
+" Language:		Adrien's Mysterious Assembler? (In Progress)
+
+" quit when a syntax file was already loaded
+if exists("b:current_syntax")
+  finish
+endif
+
+let s:cpo_save = &cpo
+set cpo&vim
+
+syn case ignore
+
+syn match asmIdentifier		"[a-z_][a-z0-9_]*"
+
+syn match asmNumber		"\<[0-9A-F]\+\>"	 display
+
+syn match asmRegister		"\<R\d\>"
+syn match asmRegister		"\<R1[0-5]\>"
+syn match asmRegister		"\<LR\>"
+syn match asmRegister		"\<PC\>"
+
+syn match asmInstruction	"nop"
+syn match asmInstruction	"mov"
+syn match asmInstruction	"str"
+syn match asmInstruction	"ldr"
+syn match asmInstruction	"bcs"
+syn match asmInstruction	"bcc"
+syn match asmInstruction	"beq"
+syn match asmInstruction	"bne"
+syn match asmInstruction	"bmi"
+syn match asmInstruction	"bpl"
+syn match asmInstruction	"bvs"
+syn match asmInstruction	"bvc"
+syn match asmInstruction	"add"
+syn match asmInstruction	"sub"
+syn match asmInstruction	"mul"
+syn match asmInstruction	"and"
+syn match asmInstruction	"or"
+syn match asmInstruction	"xor"
+syn match asmInstruction	"not"
+syn match asmInstruction	"cmp"
+syn match asmInstruction	"sec"
+syn match asmInstruction	"sez"
+syn match asmInstruction	"sen"
+syn match asmInstruction	"sev"
+syn match asmInstruction	"clc"
+syn match asmInstruction	"clz"
+syn match asmInstruction	"cln"
+syn match asmInstruction	"clv"
+syn match asmInstruction	"strmv"
+syn match asmInstruction	"stregmv"
+syn match asmInstruction	"strmemv"
+syn match asmInstruction	"swap"
+
+syn match asmComment		";.*" contains=@Spell
+
+syn match asmDirective		"\.[A-Za-z][0-9A-Za-z-_]*"
```

Comment:
Is it possible to add asm labels? A label is a directive but with a column at the end

Reply:
Do labels need to be preceded with a dot, like directives?

Reply:
A label can begin with a point but it's not mandatory

Reply:
Let's say that it doesn't so it's faster and I'm afraid of shift reduce conflicts